statsd/options.go

```diff
@@ -284,6 +287,18 @@ func WithExtendedClientSideAggregation() Option {
 	}
 }
 
+// WithMaxSamplesPerContext sets the maximum number of samples that can be aggregated in a single distribution.
+// This feature should be used with `WithExtendedClientSideAggregation`. This limits the number of sample per
+// context for a distribution to a given number.
+// This will enable client side aggregation for all metrics.
+func WithMaxSamplesPerContext(maxSamplesPerDistribution int) Option {
```

Comment:
I don't think we are ok with breaking the API

Any name suggestion here ?

`WithExtendedClientSideAggregationMaxSamples` ?

Reply:
Maybe something like this in the comment:
```go
// This feature should be used with `WithExtendedClientSideAggregation`. This limits the number of sample per
// context for a distribution to a given number. It has no impact for counts and gauges since the aggregator keeps only one value per context for these types. 
// This will enable client side aggregation for all metrics.
```